### PR TITLE
[feat] Make sanitize repo return plug.Result

### LINKED ISSUE #45: Sanitize repo should return plug.result
Right now, santize_repo returns an error if something is wrong.

We also want it to return a success message if nothing goes wrong

### PR DESCRIPTION
**Pull request by tohanss**
Fix #45

### CODE REVIEW

File: tests/test_ext_commands.py [hunk 1]
```diff
@@ -202,8 +206,28 @@ def test_sanitize_repo_return_fail_when_has_untracked_files(
...
+    """
+    target_branch = "student-version"
+
+    result = execute_sanitize_repo(
+        f"--file-list {fake_repo.file_list_path} "
+        f"--repo-root {fake_repo.path} "
+        f"--target-branch {target_branch} "
+    )
+
+    expected_success = result["sanitizer"][0]
```
Comment (tohanss) [inline tests/test_ext_commands.py:227]: Even though this works, we probably want to check if any element inside `result["sanitizer"]´ has the error we're looking for

Comment (tohanss) [reply tests/test_ext_commands.py]: Gone and fixed it

Comment (tohanss) [reply tests/test_ext_commands.py]: Though now its possible that if the tests fail it is unclear what causes it

Comment (slarse) [reply tests/test_ext_commands.py]: It's fine

**Comment by slarse** [comment]
@tohanss Minor merge conflict to resolve here

**Pull request merged**